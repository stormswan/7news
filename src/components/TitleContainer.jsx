import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import TitleBackgroundImg from './TitleBackgroundImg'
import LogoContainer from './LogoContainer'

import bgImg from '../images/tile.jpg'


export default class TitleContainer extends PureComponent {
  static propTypes = {
    title: PropTypes.string.isRequired,
  }

  render() {
    return (
      <TitleContainerStyle>
        <TitleBackgroundImg src={bgImg}/>
        <LogoContainer title={this.props.title}/>
      </TitleContainerStyle>
    )
  }
}



const TitleContainerStyle = styled.div`
  position:relative;
`