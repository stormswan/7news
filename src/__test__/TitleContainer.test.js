import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import TitleContainer from '../components/TitleContainer'


test('Shapshot', () => {
  const titleContainer = renderer.create( <TitleContainer />).toJSON()
  expect(titleContainer).toMatchSnapshot()
})


test('should has style position = relative ', () => {
  const titleContainer = renderer.create( <TitleContainer />).toJSON()
  expect(titleContainer).toHaveStyleRule('position','relative')
});