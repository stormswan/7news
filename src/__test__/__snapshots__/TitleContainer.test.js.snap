// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shapshot 1`] = `
.c1 {
  width: 100%;
}

.c4 {
  margin-left: 0.85em;
  color: white;
  font-size: 3em;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  background: rgba(25,25,25,.7);
  width: 100%;
  height: 32%;
  bottom: 0;
  left: 0;
}

.c3 {
  height: 70%;
}

.c0 {
  position: relative;
}

@media (max-width:576px) {
  .c4 {
    font-size: 1.8em;
  }
}

@media (min-width:992px) {
  .c4 {
    font-size: 4em;
  }
}

@media (min-width:1200px) {
  .c4 {
    font-size: 5em;
  }
}

<div
  className="c0"
>
  <img
    className="c1"
    src="tile.jpg"
  />
  <div
    className="c2"
  >
    <img
      alt="7 news logo"
      className="c3"
      src="logo.png"
    />
    <div
      className="c4"
      color="white"
      size="3em"
    >
      Home and Away
    </div>
  </div>
</div>
`;
